lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -136,6 +137,29 @@ def flatten_extattrs(value):
     return dict([(k, v['value']) for k, v in iteritems(value)])
 
 
+def member_normalize(member_spec):
+    ''' Transforms the member module arguments into a valid WAPI struct
```

Comment:
Can you give me an example which this recursive function is trying to resolve, coz this is heavy operation and need to check if your use case can be resolved in other ways.

Reply:
@justjais 
When no values are passed to some of the options in the module, it takes the value "None" like below.
`{
	'comment': 'CreatedbyAnsible',
	'enable_ha': True,
	'node_info': [
		{
			'v6_mgmt_network_setting': None,
			'lan_ha_port_setting': {
				'mgmt_ipv6addr': None,
				'ha_port_setting': None,
				'lan_port_setting': None,
				'ha_ip_address': '192.168.1.70',
				'mgmt_lan': '192.168.1.80'
			},
			'mgmt_network_setting': None,
			'lan2_physical_setting': None
		},
		{
			'v6_mgmt_network_setting': None,
			'lan_ha_port_setting': {
				'mgmt_ipv6addr': None,
				'ha_port_setting': None,
				'lan_port_setting': None,
				'ha_ip_address': '192.168.1.71',
				'mgmt_lan': '192.168.1.81'
			},
			'mgmt_network_setting': None,
			'lan2_physical_setting': None
		}
	],
	'config_addr_type': 'IPV4',
	'router_id': 150,
	'host_name': 'member-ha.ansible-dev.com',
	'platform': 'VNIOS',
	'vip_setting': {
		'subnet_mask': '255.255.255.0',
		'gateway': '192.168.1.1',
		'address': '192.168.1.100'
	},
	'upgrade_group': 'Default'
}`
The WAPI errors out in this situation with an error that requests for the right kind of input.
In order to avoid this, this function strips all the options set to to None from input , like below:
`{
	'comment': 'CreatedbyAnsible',
	'enable_ha': True,
	'node_info': [
		{
			'lan_ha_port_setting': {
				'ha_ip_address': '192.168.1.70',
				'mgmt_lan': '192.168.1.80'
			}
		},
		{
			'lan_ha_port_setting': {
				'ha_ip_address': '192.168.1.71',
				'mgmt_lan': '192.168.1.81'
			}
		}
	],
	'config_addr_type': 'IPV4',
	'router_id': 150,
	'host_name': 'member-ha.ansible-dev.com',
	'platform': 'VNIOS',
	'vip_setting': {
		'subnet_mask': '255.255.255.0',
		'gateway': '192.168.1.1',
		'address': '192.168.1.100'
	},
	'upgrade_group': 'Default'
}`